### PR TITLE
Document autoboot.txt and TRYBOOT_A_B mode

### PR DESCRIPTION
**Pull request by timg236**

### CODE REVIEW

**Review by aallan — APPROVED** [review]
LGTM. Ready to merge?

File: documentation/asciidoc/computers/config_txt/autoboot.adoc [hunk 1]
```diff
@@ -0,0 +1,40 @@
+== `autoboot.txt`
+
+`autoboot.txt` is an optional configuration file that can be used to specify the
+`boot_partition` number. This is sometimes used with NOOBS to bypass the boot menu
+selection and boot a specific partition.
+In the latest firmware this can also be used in conjunction with the `tryboot` feature to implement A/B booting for OS upgrades.
```
Comment (lurch) [inline documentation/asciidoc/computers/config_txt/autoboot.adoc:6]: Should "the latest firmware" be changed to "firmware newer than XXXX" ?

File: documentation/asciidoc/computers/configuration/device-tree.adoc [hunk 2]
```diff
@@ -850,18 +854,21 @@ This bit field, which describes the features supported by the current bootloader
 | xref:remote-access.adoc#network-boot-your-raspberry-pi[Network boot]
 
 | 2
-| xref:raspberry-pi.adoc#fail-safe-os-updates-tryboot[TRYBOOT]
+| xref:raspberry-pi.adoc#fail-safe-os-updates-tryboot[TRYBOOT_A_B] mode.
 
 | 3
-| xref:raspberry-pi.adoc#usb-mass-storage-boot[USB boot] using the BCM2711 USB host controller.
+| xref:raspberry-pi.adoc#fail-safe-os-updates-tryboot[TRYBOOT]
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/device-tree.adoc:860]: Wow, have the bit-positions actually changed, or was the original documentation incorrect?

**Comment by lurch** [comment]
https://github.com/raspberrypi/rpi-eeprom/pull/446 implies that this is currently only available in beta firmware images?

File: documentation/asciidoc/computers/configuration/device-tree.adoc (hunk 2 above)
Comment (timg236) [reply documentation/asciidoc/computers/configuration/device-tree.adoc:860]: The original documentation was wrong

File: documentation/asciidoc/computers/config_txt/autoboot.adoc (hunk 1 above)
Comment (timg236) [reply documentation/asciidoc/computers/config_txt/autoboot.adoc:6]: I'll remove 'latest firmware' ... with the bootloader docs we found this just bit-rotted. I think the place for that sort of stuff is on the PIP e.g. For A/B booting you will need Bullseye release X or newer.

**Comment by timg236** [comment]
> [raspberrypi/rpi-eeprom#446](https://github.com/raspberrypi/rpi-eeprom/pull/446) implies that this is currently only available in beta firmware images?

True and one day that will be stable/default release but since there's no tracking between docs  and repos it's easier to handle that on the forums.

**Comment by lurch** [comment]
The only reason I mentioned that, was because if people see it in the published docs, they might think that means that it's available in the stable release?

**Comment by timg236** [comment]
> The only reason I mentioned that, was because if people see it in the published docs, they might think that means that it's available in the stable release?

It will be in stable soon enough, probably next week

File: documentation/asciidoc/computers/config_txt/autoboot.adoc (hunk 1 above)
Comment (timg236) [reply documentation/asciidoc/computers/config_txt/autoboot.adoc]: Updated to remove 'latest firmware'

We are now just waiting for the rpi-update 'master' firmware release. Once that's done I'll ping this forum thread to direct early adopters to the bleeding edge software with added docs :)
https://forums.raspberrypi.com/viewtopic.php?p=2046776&hilit=tryboot#p2046776

**Comment by aallan** [comment]
Merging.

**Pull request merged**